src/shell/layer.rs

```diff
@@ -0,0 +1,110 @@
+/// WLR layer abstraction
+use std::cell::Cell;
+use std::rc::Rc;
+
+use wayland_client::{
+    protocol::{wl_output, wl_surface},
+    Attached, Main,
+};
+
+use wayland_protocols::wlr::unstable::layer_shell::v1::client::{
+    zwlr_layer_shell_v1, zwlr_layer_surface_v1,
+};
+
+pub use wayland_protocols::wlr::unstable::layer_shell::v1::client::{
+    zwlr_layer_shell_v1::Layer, zwlr_layer_surface_v1::Anchor,
+};
+
+/// Render event
+
+#[derive(PartialEq, Copy, Clone)]
+pub enum RenderEvent {
+    /// Surface wants a reconfiguration/configuration
+    Configure {
+        /// The new width of the surface
+        width: u32,
+
+        /// The new height of the surface
+        height: u32,
+    },
+
+    /// Surface has closed and wants to be closed here also.
+    Closed,
+}
+
+/// A struct representing the layer surface (wlr)
+
+pub struct LayerSurface {
+    /// The raw wl_surface
+    pub surface: wl_surface::WlSurface,
+
+    pub(crate) layer_surface: Main<zwlr_layer_surface_v1::ZwlrLayerSurfaceV1>,
```

Comment:
This is fine for now, we can always introduce a `LayerInner` when ext-layer-shell is initiated into wayland-protocols